src/links.ts

```diff
@@ -63,7 +63,23 @@ export function getLinks(source: string, baseUrl: string): ParsedUrl[] {
   const sanitized = links
     .filter(link => !!link)
     .map(link => parseLink(link, realBaseUrl));
-  return sanitized;
+  // Filter out the links with the same url.href like
+  // /page /page#fragment1 /page#fragment2
+  const uniqueHref = new Set();
+  const uniqueLinks: ParsedUrl[] = [];
+  sanitized.forEach(link => {
+    // ignore the links without a url
+    if (!link.url) {
+      uniqueLinks.push(link);
+    } else {
+      // if the url is existing, then we will not add this link to the list.
+      if (!uniqueHref.has(link.url.href)) {
```

Comment:
Instead of going through all of this - should we just always blank out the fragment/hash?

Reply:
yes, url of /page will be added first into the set and url of /page#fragment will be ignored. you mean filtering out links based on link itself?  
`/page.#fragment` `/page#.fragment` are two cases I have seen, not sure if it's safe to parse the link and detect `#` to do the same logic.